components/eventing-controller/pkg/handlers/jetstream.go

```diff
@@ -142,11 +149,8 @@ func (js *JetStream) SyncSubscription(subscription *eventingv1alpha1.Subscriptio
 	for _, subject := range subscription.Status.CleanEventTypes {
 		jsSubKey := js.GenerateJsSubKey(subject, subscription)
 
-		if js.conn.Status() != nats.CONNECTED {
-			if err := js.Initialize(js.connClosedHandler); err != nil {
-				log.Errorw("reset JetStream connection failed", "status", js.conn.Stats(), "error", err)
-				return err
-			}
+		if err := js.checkJetStreamConnection(log); err != nil {
```

Comment:
(nit) we are checking it on line 104, so we don't need to check it again in the same function, IMO?